_posts/2024-02-06-openttd-truetype-fonts.md

```diff
@@ -0,0 +1,90 @@
+I went through a heavily iterative process; drawing a character on the pixel grid, checking how it rasterised at 1x size with no anti-aliasing, checking where it differed from the original font, revising exact curve shapes, sub-pixel positioning of the ends of strokes etc., and repeating.
+The end result is not your typical sans-serif, and has interesting quirks like the very vertical sides of "o" and the square caps on "v and "w".
+
+## The challenges
+
+![Czech language window demonstrating diacritics]({% link /static/img/post_2024-02-06-openttd-truetype-fonts/diacritics.png %})
+
+The last big challenge was fitting everything in the very narrow line height.
+The OpenTTD GUI has a very dense text layout, originally designed to fit many informative windows into a tiny screen.
+There's absolutely no space between lines: the bottom of a "y" touches the top of a "T" on the line below.
+This means characters with diacritics (accents, etc.) require a special short letter shape, which the diacritic is then placed on.
+This was hard for the serif, very hard for the sans-serif, and extremely hard for the small font!
+I had to cheat with the small font, it's a 7-pixel tall font pretending to be 6 pixels tall, but mostly works.
+
+## The opportunities!
+
+![Signs saying latin, greek and cyrillic alphabet]({% link /static/img/post_2024-02-06-openttd-truetype-fonts/alphabets.png %})
+
+Having spent the time to make these fonts it opened up a range of options for improvements.
+OpenTTD has a very international player base, with many players using languages which use the Cyrillic or Greek alphabets.
+Having made the Latin alphabet, it was (relatively) easy to make Cyrillic and Greek alphabets, with full character coverage of all the OpenTTD translations.
+It was also (relatively) easy to make a monospaced font based on the sans-serif font, for the in-game Readmes, Changelogs, etc.
+
+Looking to the future, these fonts are complete for all European languages, but it enables easy future development.
+Having dedicated fonts makes it easier to support more languages - we can and guarantee a font is available with the necessary characters.
+There's also the possibility to add more currency symbols.
+For now, the fonts just support European scripts, but there is the possibility of future support for other alphabetic and abjad scripts.
+I'm afraid CJK scripts are probably beyond my ability though!
+
+## In closing
+
+Overall, I'm very happy with the result and I hope you like the fonts too.
+They're included with OpenTTD in version 14.0 - just make sure they're enabled in Game Options > Graphics, then toggle new vs. old fonts with the "Use traditional sprite font" button.
+The font will look best if you set the interface scaling to a round number, exactly 1x, 2x etc.
+
+![Toggling the TrueType vs. sprite font setting]({% link /static/img/post_2024-02-06-openttd-truetype-fonts/toggle.gif %})
+
+If you _really_ like the fonts then you can download and install them.
```

Comment:
This makes it sound you need to download the fonts to use it in OpenTTD, if you skip over most of the text. So maybe add some words that it is not about that. Not sure how ... "then you can download and install them to use in your local Word processor", or something.

Your next sentence does make it more clear, but I also know how our readers are: they skip every other word :P

Reply:
Good point, I'll rephrase this.